### PR TITLE
React Send UI Improvments

### PR DESCRIPTION
**Pull request by yougotwill**
- When sending a reaction in 1-1 or closed group conversation update the local state triggering a UI update when the message is sent and ignore the message response unless it is a reaction removal. This improves the perceived responsiveness of reacting to a message.
- Added additional reactor information (specifically the number of them) to the ReactListModal for open groups.

### CODE REVIEW

File: ts/components/dialog/ReactListModal.tsx [hunk 1]
```diff
@@ -159,6 +159,26 @@ const ReactionSenders = (props: ReactionSendersProps) => {
...
+      {count > SOGSReactorsFetchCount + 1
+        ? window.i18n('reactionListCountPlural', [
+            window.i18n('otherPlural', [String(count - SOGSReactorsFetchCount)]),
+          ])
+        : window.i18n('reactionListCountSingular', [
+            window.i18n('otherSingular', [String(count - SOGSReactorsFetchCount)]),
+          ])}
+    </StyledCountText>
+  );
+};
```
Comment (Bilb) [inline ts/components/dialog/ReactListModal.tsx:180]: looks clean :+1:

File: ts/components/dialog/ReactListModal.tsx [hunk 2]
```diff
@@ -230,7 +252,20 @@ export const ReactListModal = (props: Props): ReactElement => {
     if (senders.length > 0 && (!reactionsMap[currentReact]?.senders || isEmpty(_senders))) {
       setSenders([]);
     }
-  }, [currentReact, me, reaction, msgProps?.sortedReacts, reactionsMap, senders]);
+
+    if (reactionsMap[currentReact]?.count && !isEqual(count, reactionsMap[currentReact]?.count)) {
```
Comment (Bilb) [inline ts/components/dialog/ReactListModal.tsx:256]: not sure why you need the isEqual here, those are just number right?

File: ts/receiver/dataMessage.ts [hunk 3]
```diff
@@ -321,7 +321,11 @@ async function handleSwarmMessage(
     // this call has to be made inside the queueJob!
     // We handle reaction DataMessages separately
     if (!msgModel.get('isPublic') && rawDataMessage.reaction) {
-      await handleMessageReaction(rawDataMessage.reaction, msgModel.get('source'));
+      await handleMessageReaction(
+        rawDataMessage.reaction,
+        msgModel.get('source'),
+        msgModel.get('source') === UserUtils.getOurPubKeyStrFromCache()
```
Comment (Bilb) [inline ts/receiver/dataMessage.ts:327]: you could also used `isUsFromCache(msgModel.get('source'))`

File: ts/components/dialog/ReactListModal.tsx (hunk 2 above)
Comment (yougotwill) [reply ts/components/dialog/ReactListModal.tsx]: Fixed + 1

File: ts/receiver/dataMessage.ts (hunk 3 above)
Comment (yougotwill) [reply ts/receiver/dataMessage.ts:327]: Nice catch!

**Review by Bilb — APPROVED** [review]

**Pull request merged**